src/proto/FunctionRpc.proto

```diff
@@ -698,3 +701,30 @@ message ModelBindingData
 message CollectionModelBindingData {
   repeated ModelBindingData model_binding_data = 1;
 }
+
+// Retry policy which the worker sends the host when the worker indexes
+// a function.
+message RpcRetryOptions
```

Comment:
So If a function definition is decorated with this information, will the retrycontext populated as part of invocation request will use this info instead of our existing `RetryContext` prop value (which is populated from host.json?) ?

Reply:
I wonder we should rename the prop/type to be more clear that this is something function authors may define on the function level?

Reply:
The invocation request is created [here](https://github.com/Azure/azure-functions-host/blob/7164e0e6273a9ded239e154f6257082ae19b2d16/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L750) in `GrpcWorkerChannel` and the `RetryContext` field of the invocation request is populated through an extension method [here](https://github.com/Azure/azure-functions-host/blob/7164e0e6273a9ded239e154f6257082ae19b2d16/src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs#L251) that pulls from an `ExecutionContext` to get retry information.

I believe the step in `ExecutionContext` creation that populates retry information is [here](https://github.com/Azure/azure-functions-host/blob/7164e0e6273a9ded239e154f6257082ae19b2d16/src/WebJobs.Script/Description/FunctionDescriptorProvider.cs#L237) and pulls from `FunctionMetadata`, which uses info from `RpcRetryOptions`. Does that answer your question @kshyju?

Reply:
@kshyju While I think it's true that only trigger types that have function-level retry should be using this path, I'm not sure about giving this type a different name (like `FunctionLevelRetryOptions`) because it's still a representation of the webjobs/host `RetryOptions`. Happy to hear suggestions from others - it's an easy change at this stage.

As a middle-ground solution I can also revise the comment to give a more detailed explanation of this type and its usage?